### PR TITLE
Guide: Import a Bearer Did

### PR DESCRIPTION
**Pull request by blackgirlbytes**
https://deploy-preview-1628--tbd-website-developer.netlify.app/docs/web5/decentralized-identifiers/how-to-reuse-a-did

This pull request introduces a new guide for importing Decentralized Identifiers (DIDs) and includes corresponding test cases in JavaScript, Kotlin, and Swift. The guide explains how to export and import DIDs, ensuring users can maintain their identity across different platforms.

### Documentation:

* **New Guide for Importing DIDs**: Added a detailed guide on how to export and import DIDs, including code snippets for JavaScript, Kotlin, and Swift. (`site/docs/web5/decentralized-identifiers/how-to-import-did.mdx`)

### Test Cases:

* **JavaScript Tests**: Added tests to verify the export and import functionality of DIDs using the `DidDht` and `DidJwk` methods. (`site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js`)
* **Kotlin Tests**: Implemented test cases for exporting and importing DIDs using `DidDht`, `DidJwk`, and `BearerDid` classes. (`site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToImportDidTest.kt`)
* **Swift Tests**: Added tests to ensure the correct export and import of `DIDJWK` using the `InMemoryKeyManager`. (`site/testsuites/testsuite-swift/Tests/DevSiteTestSuiteTests/web5/build/decentralizedidentifiers/HowToImportDidTests.swift`)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205459115459785
  - https://app.asana.com/0/0/1208314295097366

### CODE REVIEW

**Comment by acekyd** [comment]
Nice! I was planning a short to cover this as it came up during the hackathon - but I ran into a question I want to pose now. For the title, would it be better to use something like "Managing DIDs or reusing DIDs etc since it covers both exporting and importing - for the sole purpose of reusing an existing DID 🤔

**Comment by blackgirlbytes** [comment]
> Nice! I was planning a short to cover this as it came up during the hackathon - but I ran into a question I want to pose now. For the title, would it be better to use something like "Managing DIDs or reusing DIDs etc since it covers both exporting and importing - for the sole purpose of reusing an existing DID 🤔

oh okay! That's fair. Angie assigned a ticket to me called Import DIDs so I just named it that LOL but that's fair/a clearer name

File: site/docs/web5/decentralized-identifiers/how-to-import-did.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+
+
+# Import a DID
+
+In this guide, you will learn how to take your [Decentralized Identifier (DID)](/docs/web5/decentralized-identifiers/what-are-dids) with you to other apps, allowing you to keep your identity consistent everywhere across various platforms. 
+
+## Export DID
+You will start by exporting your DID. This method converts your [Bearer DID](/docs/web5/decentralized-identifiers/how-to-create-did/#create-did) into a [Portable DID](/docs/web5/decentralized-identifiers/how-to-create-did/#portable-did), a serialized and static representation suitable for storage or transfer.
```
Comment (angiejones) [inline site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:14]: we should add glossary terms for Bearer DID and Portable DID and link to those

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { test, expect, describe } from 'vitest';
+
+describe('Import and export DIDs', () => {
+    test('Exports a Bearer DID to a Portable DID', async () => {
+      const didDht = await DidDht.create({ options: { publish: true } });
+      const didJwk = await DidJwk.create();
+  
+      // :snippet-start: exportDidJs
+      // export did:dht DID
+      const portableDhtDid = await didDht.export();
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js:13]: in these 2 lines, it's not clear that didDht is a bearer did. can you rename these variables up top to `dhtBearerDid` and `jwkBearerDid`?

File: site/docs/web5/decentralized-identifiers/how-to-import-did.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  snippets={[
+    { snippetName: 'exportDidJs', language: 'JavaScript'},
+    { snippetName: 'exportDidKt', language: 'Kotlin' },
+        { snippetName: 'exportDidSwift', language: 'Swift' }
+  ]}
+/>
+
+
+## Import DID
+Now, you can import your Portable DID to a desired location using the `import` method.
```
Comment (angiejones) [inline site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:30]:
```suggestion
You can import your Portable DID to an application using the `import` method. 
```

File: site/docs/web5/decentralized-identifiers/how-to-import-did.mdx (hunk 1 above)
Comment (angiejones) [inline site/docs/web5/decentralized-identifiers/how-to-import-did.mdx:14]: can you specify what exactly gets returned? is it a JSON?

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js (hunk 2 above)
Comment (blackgirlbytes) [reply site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-import-did.test.js]: done

File: site/docs/web5/decentralized-identifiers/how-to-import-did.mdx (hunk 1 above)
Comment (blackgirlbytes) [reply site/docs/web5/decentralized-identifiers/how-to-import-did.mdx]: added!

**Comment by blackgirlbytes** [comment]
## Updates

- Changed the title from Import a DID to Reuse a DID based on @acekyd's suggestions
- Added Bearer DID and Portable DID to glossary, updated mentions of it based on @angiejones's suggestions.

**Review by angiejones — APPROVED** [review]

**Pull request merged**